lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -524,21 +561,20 @@ def _consume_file(read_from, write_to=None):
 
 
 def _normalize_galaxy_yml_manifest(
-        galaxy_yml,  # type: dict[str, t.Union[str, list[str], dict[str, str], None, t.Type[Sentinel]]]
-        b_galaxy_yml_path,  # type: bytes
-        require_build_metadata=True,  # type: bool
-):
-    # type: (...) -> dict[str, t.Union[str, list[str], dict[str, str], None, t.Type[Sentinel]]]
-    galaxy_yml_schema = (
-        get_collections_galaxy_meta_info()
-    )  # type: list[dict[str, t.Any]]  # FIXME: <--
-    # FIXME: 👆maybe precise type: list[dict[str, t.Union[bool, str, list[str]]]]
-
-    mandatory_keys = set()
-    string_keys = set()  # type: set[str]
-    list_keys = set()  # type: set[str]
-    dict_keys = set()  # type: set[str]
-    sentinel_keys = set()  # type: set[str]
+        galaxy_yml: dict[
+            str,
+            str | list[str] | dict[str, str] | None | t.Type[Sentinel],
```

Comment:
The `manifest` boolean option `omit_default_directives` is not represented here.

```suggestion
            str | list[str] | dict[str, str | bool] | None | t.Type[Sentinel],
```

Reply:
Yeah, I temporarily gave up on this function as it's non-trivial to address (hence the cast at the end). It should definitely be refactored more in the future. But I didn't want to make the PR scope too big. I'll accept this suggestion but it'll only cover a little bit of what needs to be improved.